pyrobosim/pyrobosim/navigation/occupancy_grid.py

```diff
@@ -97,6 +98,19 @@ def save_to_file(self, folder, filename="world_map"):
         with open(yaml_file, "w") as f:
             yaml.dump(yaml_dict, f, sort_keys=False, default_flow_style=None)
 
+    def world_to_grid(self, x, y):
+        x_grid = math.floor((x - self.origin[0]) / self.resolution)
+        y_grid = math.floor((y - self.origin[1]) / self.resolution)
+        return (x_grid, y_grid)
+
+    def grid_to_world(self, x, y):
+        x_world = (x * self.resolution) + self.origin[0]
+        y_world = (y * self.resolution) + self.origin[1]
+        return (x_world, y_world)
+
+    def is_occupied(self, x, y):
```

Comment:
This should handle out of bounds checking in case x and y aren't actually part of the grid, else you will throw errors. Easiest to have it return true if out of bounds.

Reply:
It will be good to print an "Out of bounds" warning as well.

Reply:
If so, use the `warnings` library so it doesn't spam the terminal if printed multiple times. I use that in other places in pyrobosim, so should be findable.

Also to avoid such warnings in the A* planner, would recommend also checking bounds at the `expand()` level of the planner